src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToSearchResultProvider.cs

```diff
@@ -18,12 +19,56 @@ internal sealed partial class NavigateToSearchResultProvider : INavigateToSearch
     {
         public async Task<IEnumerable<INavigateToSearchResult>> SearchProjectAsync(Project project, string searchPattern, CancellationToken cancellationToken)
         {
-            var results = await NavigateToSymbolFinder.FindNavigableDeclaredSymbolInfos(project, searchPattern, cancellationToken).ConfigureAwait(false);
+            var results = await FindNavigableDeclaredSymbolInfos(project, searchPattern, cancellationToken).ConfigureAwait(false);
             var containsDots = searchPattern.IndexOf('.') >= 0;
             return results.Select(r => ConvertResult(containsDots, r));
         }
 
-        private INavigateToSearchResult ConvertResult(bool containsDots, ValueTuple<DeclaredSymbolInfo, Document, IEnumerable<PatternMatch>> result)
+        private static async Task<IEnumerable<ValueTuple<DeclaredSymbolInfo, Document, IEnumerable<PatternMatch>>>> FindNavigableDeclaredSymbolInfos(
+            Project project, string pattern, CancellationToken cancellationToken)
+        {
+            using (var patternMatcher = new PatternMatcher(pattern, allowFuzzyMatching: true))
+            {
+                var result = new List<ValueTuple<DeclaredSymbolInfo, Document, IEnumerable<PatternMatch>>>();
```

Comment:
Every time a generic is closed with more than three `>`s I lose one night of sleep. :smile: